syntax = "proto3";

package deployagent;

service AgentDeployService {
  rpc DeployAgent(DeployAgentRequest) returns (DeployAgentResponse) {}
  rpc DeleteAgent(DeleteAgentRequest) returns (DeleteAgentResponse) {}
}

message RegistryCredentials {
  string username = 1;
  string password = 2;
}

// Deploy an agent as a Fly.io application
message DeployAgentRequest {
  string name = 1;
  string image_url = 2;  // Optional: external registry image. If empty, uses latest build.
  int64 agent_id = 3;    // Agent ID for naming the Fly app
  optional RegistryCredentials registry_credentials = 4; // Optional credentials for private registry
}

message DeployAgentResponse {
  enum Status {
    SUCCESS = 0;
    ERROR = 1;
  }

  Status status = 1;
  string message = 2;
  string app_name = 3;           // The Fly.io app name
  string deployed_image_url = 4;  // The deployed image URL
}

// Delete an agent deployment
message DeleteAgentRequest {
  string name = 1;
  int64 agent_id = 2;  // Agent ID to identify the Fly app
}

message DeleteAgentResponse {
  enum Status {
    SUCCESS = 0;
    ERROR = 1;
  }
  
  Status status = 1;
  string message = 2;
}

